front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -77,30 +80,69 @@ const CurveSpreadsheet = ({
         ...(isDefaultCurve ? { default_curve: formattedCurve } : {}),
       },
     };
+
+    // Set the updated effort curves
     setEffortCurves(updatedEffortCurve);
   };
 
-  const orderSpreadsheetValues = () => {
-    const orderedValuesByVelocity = spreadsheetCurve.sort((a, b) => {
-      // if a row has a max_effort, but no speed, it should appear at the top of the table
-      if (b[0] && b[0].value === '') return 1;
-      return Number(a[0]?.value) - Number(b[0]?.value);
-    });
-    updateRollingStockCurve(orderedValuesByVelocity);
-  };
+  const spreadsheetCurve = useMemo(() => {
+    const { speeds, max_efforts } = selectedCurve.curve;
+    let filledDataSheet: DataSheetCurve[] = [];
+
+    if (speeds && max_efforts) {
+      filledDataSheet = max_efforts.map((effort, index) => ({
+        speed: speeds[index] !== undefined ? Math.round(msToKmh(speeds[index]!)) : '',
+        // Effort needs to be displayed in kN
+        effort: effort !== undefined ? effort / 1000 : '',
+      }));
+    }
```

Comment:
max_efforts and speeds are always defined, so no need for the test :+1: 
```suggestion
    let filledDataSheet: DataSheetCurve[] = max_efforts.map((effort, index) => ({
      speed: speeds[index] !== null ? Math.round(msToKmh(speeds[index]!)) : null,
      // Effort needs to be displayed in kN
      effort: effort !== null ? effort / 1000 : null,
    }));
```

Reply:
done